src/audio/dai.c

```diff
@@ -66,6 +93,59 @@ struct dai_data {
 	uint64_t wallclock;	/* wall clock at stream start */
 };
 
+static void dai_atomic_trigger(void *arg, enum notify_id type, void *data);
+
+static int dai_add_group(struct comp_dev *dev, uint32_t group_id)
```

Comment:
Should be named dai_add_to_group()? Actually this function does both, creates(adds) a group and adds dai to the group. So how about creating dai_group API for group mgmt similar to the one for dai mgmt defined in src/lib/dai.c? So that dai_groups in sof is manipulated via that API? dai_group_get(..., flag = DG_CREAT /* create if does not exist *?), dai_group_add() etc. ?